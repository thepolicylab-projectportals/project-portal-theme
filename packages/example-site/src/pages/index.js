import React from "react"
import { Layout } from "@thepolicylab-projectportals/gatsby-theme-project-portal/src/layouts"

const Index = () => {
  return (
    <Layout title={"the title"} description={"the description"}>
      <h1>Heading 1</h1>
    </Layout>
  )
}

export default Index
