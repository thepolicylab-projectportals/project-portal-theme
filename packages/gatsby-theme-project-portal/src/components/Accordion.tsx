import React from "react"
import { Disclosure } from "@headlessui/react"
import { FaMinus, FaPlus } from "react-icons/fa"
import { MarkdownText } from "."

interface AccordionProps {
  title: string
  text: string
}

export const Accordion: React.FC<AccordionProps> = ({ title, text }) => {
  return (
    <div className="w-full my-6">
      <Disclosure>
        {({ open }) => (
          <>
            <Disclosure.Button className="flex items-center justify-between w-full px-4 py-3 text-left bg-navbar">
              <span className="text-h4 font-bold">{title}</span>
              {open ? <FaMinus /> : <FaPlus />}
            </Disclosure.Button>
            <Disclosure.Panel className="text-body p-4 w-9/10">
              <MarkdownText text={text} />
            </Disclosure.Panel>
          </>
        )}
      </Disclosure>
    </div>
  )
}
