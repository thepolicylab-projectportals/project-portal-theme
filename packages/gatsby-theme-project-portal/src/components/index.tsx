export { BackIcon } from "./BackIcon"
export { CollaboratorDetails } from "./CollaboratorDetails"
export { ForwardIcon } from "./ForwardIcon"
export { DevelopmentBanner } from "./DevelopmentBanner"
export { ProjectStatus } from "./ProjectStatus"
export { MarkdownText } from "./MarkdownText"
export { Feature } from "./Feature"
export { ShareProject } from "./ShareProject"
export { SectionOfItem } from "./SectionofItem"
export { Card } from "./Card"
export type { CardProps } from "./Card"
export { Cards } from "./Cards"
export { NavbarLayout, Navbar } from "./Navbar"
export { BottomBannerLayout, BottomBanner } from "./BottomBanner"
export { SiteMetadata } from "./SiteMetadata"
export { FooterLayout, Footer } from "./Footer"
export { Contact } from "./Contact"
export { ProjectTeam } from "./ProjectTeam"
export type { ContactType } from "./Contact"
export { MainContact } from "./MainContact"
