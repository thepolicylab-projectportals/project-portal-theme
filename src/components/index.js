export { Card, CardProps } from "./Card"
export { Cards } from "./Cards"
export { Feature } from "./Feature"
export { Footer } from "./Footer"
export { Hero } from "./Hero"
export { Navbar } from "./Navbar"
export { SignupForm } from "./SignupForm"
export { SiteMetadata } from "./SiteMetadata"
