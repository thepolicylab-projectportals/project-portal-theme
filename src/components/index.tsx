export { BottomBanner } from "./BottomBanner"
export { Card } from "./Card"
export { Cards } from "./Cards"
export { CollaboratorDetails } from "./CollaboratorDetails"
export { Contact } from "./Contact"
export { Feature } from "./Feature"
export { Footer } from "./Footer"
export { MainContact } from "./MainContact"
export { Navbar } from "./Navbar"
export { ProjectPage } from "./ProjectPage"
export { ProjectTeam } from "./ProjectTeam"
export { SectionOfItem } from "./SectionOfItem"
export { ShareProject } from "./ShareProject"
export { SiteMetadata } from "./SiteMetadata"

export type { CardProps } from "./Card"
export type { ContactType } from "./Contact"
export type { ProjectPageProps } from "./ProjectPage"
