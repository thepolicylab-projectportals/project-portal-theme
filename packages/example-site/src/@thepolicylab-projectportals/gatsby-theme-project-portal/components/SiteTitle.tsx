import React, { FunctionComponent } from "react"
import { GatsbyImage } from "gatsby-plugin-image"
import { SiteTitleProps } from "@thepolicylab-projectportals/gatsby-theme-project-portal/src/components"

export const SiteTitle: FunctionComponent<SiteTitleProps> = ({ image }) => {
  return (
    <>
      {image && (
        <GatsbyImage
          className="xl:inline-block logotype"
          image={image}
          alt={"nav_logo"}
        />
      )}
    </>
  )
}
