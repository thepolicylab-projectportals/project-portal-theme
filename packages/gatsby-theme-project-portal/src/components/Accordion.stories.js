import { Accordion } from "./Accordion"

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
export default {
  title: "Theme/Accordion",
  component: Accordion,
  tags: ["autodocs"],
  argTypes: {},
}

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Primary = {
  args: { title: "A title", text: "The text" },
}
