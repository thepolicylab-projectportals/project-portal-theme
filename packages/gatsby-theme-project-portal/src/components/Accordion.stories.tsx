// Ignore checks of unused exported defaults & constants as they are used implicitly by Storybook
// noinspection JSUnusedGlobalSymbols

import type { Meta, StoryObj } from "@storybook/react"
import { Accordion } from "./Accordion"

const meta: Meta<typeof Accordion> = {
  component: Accordion,
  tags: ["autodocs"],
}

export default meta

type Story = StoryObj<typeof Accordion>

export const Primary: Story = {
  args: {
    title: "What is a good text for testing a component?",
    text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
  },
}

export const LongTitle: Story = {
  args: {
    title:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    text: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
  },
}
export const NoTitle: Story = {
  args: { title: "", text: "The text" },
}

export const NoText: Story = {
  args: { title: "A title", text: "" },
}
