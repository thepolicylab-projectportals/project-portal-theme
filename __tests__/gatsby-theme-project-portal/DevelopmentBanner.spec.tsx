import React from "react"
import "@testing-library/jest-dom"

import { render, screen } from "@testing-library/react"
import { DevelopmentBanner } from "@thepolicylab-projectportals/gatsby-theme-project-portal/src/components"

describe("test DevelopmentBanner Component", () => {
  test("Should render development banner component", () => {
    const component = render(<DevelopmentBanner />)
    expect(component.container).toMatchSnapshot()
  })

  it("renders with the correct text and link", () => {
    const { getByText } = render(<DevelopmentBanner />)
    //checks that correct text is rendered
    const textElement = getByText(/This is a beta site\./)
    expect(textElement).toBeInTheDocument()
    const linkElement = getByText("We welcome feedback!")
    expect(linkElement).toHaveAttribute("href", "/contact")
  })

  it("renders a link to the contact page", () => {
    const { getByText } = render(<DevelopmentBanner />)
    const link = getByText("We welcome feedback!")
    expect(link).toBeInTheDocument()
    //checks that link text is rendered and that it is a link to "/contact"
    expect(link.getAttribute("href")).toEqual("/contact")
  })

  it("applies the correct styles to the banner", () => {
    const { getByText } = render(<DevelopmentBanner />)
    const banner = getByText(/This is a beta site\./).parentElement
    //tailwind styling
    expect(banner).toHaveClass("py-6 overflow-hidden bg-warning text-center")
  })
})
