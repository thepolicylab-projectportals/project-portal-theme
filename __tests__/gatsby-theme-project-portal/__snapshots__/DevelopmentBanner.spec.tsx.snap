// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DevelopmentBanner Component Should render development banner component 1`] = `
<div>
  <div
    class="py-6 overflow-hidden bg-warning text-center"
  >
    <h4>
      ⚠️ This is a beta site.
       
      <a
        class="underline hover:no-underline"
        href="/contact"
      >
        We welcome feedback!
      </a>
       
      ⚠️
    </h4>
  </div>
</div>
`;
